### PR TITLE
fix static analysis errors in shopwares core files

### PR DESCRIPTION
**Pull request by jochenmanz**
### 1. Why is this change necessary?
shopware has numerous wrong typehints and even missing use statements. This PR tries to eliminate some of them

### 2. What does this change do, exactly?
fixes a huge amount of wrong typehints, missing use statements, duplicate array key usage, missing parameters, unnecessary function calls, unused methods and some generic optimisations for less error prone static analysis of the shopware core

### 3. Describe each step to reproduce the issue or behaviour.
phpstorm should display most of the errors when comparing the files with the older version. For a more complete look you need to install phpstan and run a level=1 analysis. 

### 4. Please link to the relevant issues (if any).
none

### 5. Which documentation changes (if any) need to be made because of this PR?
none

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Shopware/Components/Api/Resource/Cache.php [hunk 1]
```diff
@@ -55,13 +55,13 @@ class Cache extends Resource implements ContainerAwareInterface, BatchInterface
     /**
      * Sets the Container.
      *
-     * @param Container $container
+     * @param Container $Container
      */
-    public function setContainer(Container $container = null)
+    public function setContainer(Container $Container = null)
```
Comment (mitelg) [inline engine/Shopware/Components/Api/Resource/Cache.php:60]: why the variable name in uppercase?

File: engine/Shopware/Components/Api/Resource/Resource.php [hunk 2]
```diff
@@ -98,11 +98,11 @@ public function getContainer()
     }
 
     /**
-     * @param $container
+     * @param Container $Container
      */
-    public function setContainer($container)
+    public function setContainer(Container $Container = null)
```
Comment (mitelg) [inline engine/Shopware/Components/Api/Resource/Resource.php:103]: why the variable name in uppercase?

File: engine/Shopware/Components/Emotion/Preset/PresetLoader.php [hunk 3]
```diff
@@ -57,7 +57,7 @@ public function load($presetId)
         $preset = $this->modelManager->getRepository(Preset::class)->find($presetId);
 
         if (!$preset) {
-            throw new NoResultException('The preset with id ' . $presetId . ' could not be found.');
+            throw new RuntimeException('The preset with id ' . $presetId . ' could not be found.');
```
Comment (mitelg) [inline engine/Shopware/Components/Emotion/Preset/PresetLoader.php]: this is a break, isn't it?

File: engine/Shopware/Models/Country/Repository.php [hunk 4]
```diff
@@ -277,37 +277,4 @@ public function getStatesByCountryIdQueryBuilder($countryId)
...
-
-    /**
-     * Returns an instance of \Doctrine\ORM\Query object which selects a
-     * list of countries for the passed area id.
-     *
-     * @param $stateId
-     *
-     * @return \Doctrine\ORM\Query
-     */
-    public function getStateAttributesQuery($stateId)
```
Comment (mitelg) [inline engine/Shopware/Models/Country/Repository.php]: removing public methods is a break

File: engine/Shopware/Components/Api/Resource/Resource.php (hunk 2 above)
Comment (jochenmanz) [reply engine/Shopware/Components/Api/Resource/Resource.php:103]: Shopwares ContainerAwareInterface says it needs to be uppercase :)

File: engine/Shopware/Components/Api/Resource/Cache.php (hunk 1 above)
Comment (jochenmanz) [reply engine/Shopware/Components/Api/Resource/Cache.php:60]: Shopwares ContainerAwareInterface says it needs to be uppercase :)

File: engine/Shopware/Models/Newsletter/Address.php [hunk 5]
```diff
@@ -179,22 +180,6 @@ public function getIsCustomer()
         return $this->isCustomer;
     }
 
-    /**
-     * @param int $lastMailingId
-     */
-    public function setLastMailingId($lastMailingId)
```
Comment (jochenmanz) [inline engine/Shopware/Models/Newsletter/Address.php]: Also a BC break then. But how to proceed with a function if the class property is missing?

File: engine/Shopware/Models/Newsletter/Newsletter.php [hunk 6]
```diff
@@ -467,22 +470,6 @@ public function getId()
         return $this->id;
     }
 
-    /**
-     * @return \Doctrine\Common\Collections\ArrayCollection
-     */
-    public function getOrders()
-    {
-        return $this->orders;
```
Comment (jochenmanz) [inline engine/Shopware/Models/Newsletter/Newsletter.php]: also a bc break, but unfortunatly the class proerty orders and alreadySentTo doenst exist. Should we warn via a trigger_error?

**Pull request closed without merging**